### PR TITLE
Collections refactor playbooks

### PR DESCRIPTION
**Pull request by domenicbove**
# Description

- Moves playbooks into ansible collection standard directory
- Moves tasks directory as well
- Updates all molecule scenarios to test collection playbook instead of use relative path

Part of # (https://github.com/confluentinc/cp-ansible/issues/613)

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

Ran through many of the scenarios manually, jenkins should also validate

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] Any dependent changes have been merged and published in downstream modules
- [ ] Any variable changes have been validated to be backwards compatible

### CODE REVIEW

File: molecule/kafka-connect-replicator-plain-kerberos-rhel/molecule.yml [hunk 1]
```diff
@@ -206,9 +204,9 @@ provisioner:
           sasl_protocol: plain
 
         kafka_connect_replicator_consumer_bootstrap_servers: mds-kafka-broker1:9092
-        kafka_connect_replicator_consumer_ssl_ca_cert_path: "molecule/{{scenario_name}}/generated_ssl_files/ca.crt"
-        kafka_connect_replicator_consumer_ssl_cert_path: "molecule/{{scenario_name}}/generated_ssl_files/{{inventory_hostname}}-ca1-signed.crt"
-        kafka_connect_replicator_consumer_ssl_key_path: "rmolecule/{{scenario_name}}/generated_ssl_files/{{inventory_hostname}}-key.pem"
+        kafka_connect_replicator_consumer_ssl_ca_cert_path: "{{ lookup('env', 'MOLECULE_SCENARIO_DIRECTORY') }}/generated_ssl_files/ca.crt"
+        kafka_connect_replicator_consumer_ssl_cert_path: "{{ lookup('env', 'MOLECULE_SCENARIO_DIRECTORY') }}/generated_ssl_files/{{inventory_hostname}}-ca1-signed.crt"
+        kafka_connect_replicator_consumer_ssl_key_path: "../../rmolecule/{{scenario_name}}/generated_ssl_files/{{inventory_hostname}}-key.pem"
```
Comment (domenicbove) [inline molecule/kafka-connect-replicator-plain-kerberos-rhel/molecule.yml:209]: fix this one

File: molecule/confluent-kafka-kerberos-customcerts-rhel/molecule.yml [hunk 2]
```diff
@@ -174,25 +172,25 @@ provisioner:
       zookeeper:
         # Paths are relative to all.yml
```
Comment (erikgb) [inline molecule/confluent-kafka-kerberos-customcerts-rhel/molecule.yml:173]:
```suggestion
```

Is this comment still relevant?

File: molecule/mtls-custombundle-rhel/molecule.yml [hunk 3]
```diff
@@ -113,9 +113,6 @@ platforms:
     networks:
       - name: confluent
 provisioner:
-  playbooks:
-    prepare: prepare.yml
```
Comment (erikgb) [inline molecule/mtls-custombundle-rhel/molecule.yml:117]: This appears to be a change, but Molecule might have that prepare-playbook path as the default?

**Review by erikgb — COMMENTED** [review]
Overall, this looks good. Great job! A few general comments:

- I still find a few references to `all.yml` in `test_scripts/reconfigure-basic.sh` and `test_scripts/reconfigure-certs.sh`, and I think those should be updated as well as part of this PR.
- I suggest splitting out the bugfix to the molecule paths, probably introduced in https://github.com/confluentinc/cp-ansible/pull/701, into a PR targetting the main branch. WDYT?
- Before merging this PR, I suggest an interactive rebase to ensure that the PR consists of atomic commits.

File: molecule/mtls-custombundle-rhel/molecule.yml (hunk 3 above)
Comment (domenicbove) [reply molecule/mtls-custombundle-rhel/molecule.yml:117]: ya so if you provide a playbook named "prepare.yml" this line is not needed. that playbook will get used as for the prepare phase (which is what happens). Only if that playbook had a different name is this line necessary

File: molecule/collections_converge.yml [hunk 4]
```diff
@@ -0,0 +1,2 @@
+---
+- import_playbook: confluent.platform.all
```
Comment (domenicbove) [inline molecule/collections_converge.yml:2]:
```suggestion
- import_playbook: confluent.platform.all

```

File: molecule/ccloud/molecule.yml [hunk 5]
```diff
@@ -123,6 +123,8 @@ platforms:
     networks:
       - name: confluent
 provisioner:
+  playbooks:
+    converge: converge.yml
```
Comment (domenicbove) [inline molecule/ccloud/molecule.yml:127]:
```suggestion
```

**Review by erikgb — APPROVED** [review]
LGTM! But I would like to see a successful build. @domenicbove Any updates?

**Pull request merged**